include/meshio/vectors.hpp

```diff
@@ -7,172 +7,199 @@
+
+    inline T getX() const { return elems[0]; }
+    inline T getY() const { return elems[1]; }
+    inline T getZ() const { return elems[2]; }
+    inline T getW() const { return elems[3]; }
+
+    viterator begin() { return elems.begin(); }
+    const_viterator begin() const { return elems.begin(); }
+
+  private:
+    std::vector<T> elems;
 };
 
 template<class T>
-class Vec3 {
+class Vec3 : public Vec4<T> {
   public:
-    T x,y,z;
-
-    Vec3(T pX, T pY, T pZ)
-        : x(pX), y(pY), z(pZ) {
-    }
-
-    Vec3()
-        : x(0), y(0), z(0) {
-    }
-
-    Vec3& operator+=(const Vec3& other) {
-        x += other.x;
-        y += other.y;
-        z += other.z;
-        return *this;
-    }
-
-    Vec3& operator-=(const Vec3& other) {
-        x -= other.x;
-        y -= other.y;
-        z -= other.z;
-        return *this;
-    }
-
-    Vec3& operator*=(const T pDiv) {
-        x *= pDiv;
-        y *= pDiv;
-        z *= pDiv;
-        return *this;
-    }
-
-    Vec3& operator/=(const T pDiv) {
-        T oneByPDiv = (T)1/pDiv;
-        return operator*=(oneByPDiv);
-    }
-
-    bool operator==(const Vec3& other) {
-        return ((x == other.x) && (y == other.y) && (z == other.z));
-    }
+    typedef typename std::vector<T>::iterator viterator;
+    typedef typename std::vector<T>::const_iterator const_viterator;
+
+    Vec3(T pX, T pY, T pZ) : Vec4<T>(pX, pY, pZ, 0) {}
+    Vec3() : Vec4<T>(0, 0, 0, 0) {}
+    ~Vec3() {}
+
+    std::size_t size() const { return 3; }
+    viterator end() { return --Vec4<T>::end(); }
+    const_viterator end() const { return --Vec4<T>::end(); }
 };
 
 template<class T>
-class Vec4 {
+class Vec2 : public Vec4<T> {
   public:
-    T x,y,z,w;
-
-    Vec4(T pX, T pY, T pZ, T pW)
-        : x(pX), y(pY), z(pZ), w(pW) {
-    }
-
-    Vec4()
-        : x(0), y(0), z(0), w(0) {
-    }
-
-    Vec4& operator+=(const Vec4& other) {
-        x += other.x;
-        y += other.y;
-        z += other.z;
-        w += other.w;
-        return *this;
-    }
-
-    Vec4& operator-=(const Vec4& other) {
-        x -= other.x;
-        y -= other.y;
-        z -= other.z;
-        w -= other.w;
-        return *this;
-    }
-
-    Vec4& operator*=(const T pDiv) {
-        x *= pDiv;
-        y *= pDiv;
-        z *= pDiv;
-        w *= pDiv;
-        return *this;
-    }
-
-    Vec4& operator/=(const T pDiv) {
-        T oneByPDiv = (T)1/pDiv;
-        return operator*=(oneByPDiv);
-    }
-
-    bool operator==(const Vec4& other) {
-        return ((x==other.x) && (y==other.y) && (z==other.z) && (w==other.w));
-    }
+    typedef typename std::vector<T>::iterator viterator;
+    typedef typename std::vector<T>::const_iterator const_viterator;
+
+    Vec2(T pX, T pY) : Vec4<T>(pX, pY, 0, 0) {}
+    Vec2() : Vec4<T>(0, 0, 0, 0) {}
+    ~Vec2() {}
+
+    std::size_t size() const { return 2; }
+    viterator end() { return (Vec4<T>::end()-2); }
+    const_viterator end() const { return (Vec4<T>::end()-2); }
```

Comment:
Not sure, but if we are not using this yet. Lets just comment it out and re-enable it when it can be used. Don't delete it though, just comment it out.